sdk/src/main/scala/io/horizen/AbstractSidechainApp.scala

```diff
@@ -70,6 +70,7 @@ abstract class AbstractSidechainApp
 
   private val closableResourceList = mutable.ListBuffer[AutoCloseable]()
   protected val sidechainTransactionsCompanion: DynamicTypedSerializer[TX, TransactionSerializer[TX]]
+  protected val terminationTimeout: FiniteDuration = Duration(30, TimeUnit.SECONDS)
```

Comment:
I think in the most of the cases 30seconds must be ok.
But what if the node owner set in the config file the `settings.sparkzSettings.restApi.timeout` duration bigger than 30.
If I'm not wrong, a lot of `await` cases depend on that parameter (forger, submitter, etc)

To be even more safe we may set `terminationTimeout` as a `Min(restApi.timeout, 30s)`
What do you think?

Reply:
After some more discussion, we came to the conclusion that there is no healthy situation where we want to wait for node termination for more than 30 seconds. Even if the rest timeout is bigger, the fact that some actor has to wait until it's expiration indicates some problematic behavior.